Description: update for additional NM device types
 This patch adds support for mobile broadband device types.
 It should be backwards compatible with 0.6, 0.7, and 0.8 as well as using
 the new device types in 0.9.
Author: danielcbwilliams@gmail.com
Bug: http://code.google.com/p/google-gadgets-for-linux/issues/detail?id=378

diff -up google-gadgets-for-linux-0.11.2/extensions/linux_system_framework/network.cc.devtypes google-gadgets-for-linux-0.11.2/extensions/linux_system_framework/network.cc
--- google-gadgets-for-linux-0.11.2/extensions/linux_system_framework/network.cc.devtypes	2011-03-25 11:04:20.697374611 -0500
+++ google-gadgets-for-linux-0.11.2/extensions/linux_system_framework/network.cc	2011-03-25 11:08:01.732611291 -0500
@@ -26,6 +26,11 @@ namespace linux_system {
 static const int kDeviceTypeUnknown = 0;
 static const int kDeviceTypeEthernet = 1;
 static const int kDeviceTypeWifi = 2;
+static const int kDeviceTypeOldGsm = 3;
+static const int kDeviceTypeOldCdma = 4;
+static const int kDeviceTypeBluetooth = 5;
+static const int kDeviceTypeWimax = 7;
+static const int kDeviceTypeModem = 8;
 
 // 0.6, 0.7, and 0.8 connected state
 static const int kOldConnected = 3;
@@ -171,6 +176,15 @@ void Network::Update() {
           } else if (type == kDeviceTypeWifi) {
             connection_type_ = CONNECTION_TYPE_NATIVE_802_11;
             physcial_media_type_ = PHYSICAL_MEDIA_TYPE_NATIVE_802_11;
+          } else if (type == kDeviceTypeBluetooth) {
+            connection_type_ = CONNECTION_TYPE_WIRELESS_WAN;
+            physcial_media_type_ = PHYSICAL_MEDIA_TYPE_BLUETOOTH;
+          } else if (type == kDeviceTypeModem ||
+                     type == kDeviceTypeOldGsm ||
+                     type == kDeviceTypeOldCdma ||
+                     type == kDeviceTypeWimax) {
+            connection_type_ = CONNECTION_TYPE_WIRELESS_WAN;
+            physcial_media_type_ = PHYSICAL_MEDIA_TYPE_WIRELESS_WAN;
           } else {
             connection_type_ = CONNECTION_TYPE_UNKNOWN;
             physcial_media_type_ = PHYSICAL_MEDIA_TYPE_UNSPECIFIED;
